miniapp/src/main/java/com/rakuten/mobile/miniapp/miniapp/MiniAppDownloader.kt

```diff
@@ -0,0 +1,11 @@
+package com.rakuten.mobile.miniapp.miniapp
+
+import com.rakuten.mobile.miniapp.api.ApiClient
+import com.rakuten.mobile.miniapp.storage.MiniAppStorage
+
+@SuppressWarnings("UseDataClass")
+internal class MiniAppDownloader(
```

Comment:
I was thinking maybe this can just be named `Downloader`? It's already in a package named `miniapp` so the name `MiniAppDownloader` seems redundant. And same with the other classes in this package. What do you think?

- [ ] Consider renaming MiniAppDownloader, MiniAppLister, and MiniAppStatus to Downloader, Lister, Status

Reply:
It is a tricky one actually. In Java, it used to be a single public class per file. In Kotlin, we can have multiple of those in one file. File/Class names used to be descriptive and contextual of what they tend to do. Package names in the namespace augmented to provide the context for a file's purpose and grouping.

For faster intuitivity, I think it is better to have a file/class that depicts its purpose by name and package's information context as a secondary supporting artifact. 

It's difficult to understand the purpose of Lister and Status just by its name (let say who's status or what type of lister). For `MiniAppDownloader`, I thought to rely on the secondary context, `legacy`. Further, It might make use of a separate downloader and update checker class.
I think we can rename the `miniapp` package to `base` or alike. What do you think?

Reply:
Yeah, makes sense I think. But if we name them `MiniAppDownloader`, etc. then I think they don't need to be in a sub-package. Maybe they could just be in the root package at `com.rakuten.tech.mobile.miniapp.MiniAppDownloader`. These classes will have most of the core business rules, so maybe it makes more sense to be at the root?